src/value/owned.rs

```diff
@@ -158,25 +179,23 @@ impl Struct for OwnedStruct {
         )
     }
 
-    fn get(&self, field_name: &str) -> Option<&Self::Element> {
-        match self.text_fields.get(field_name)?.last() {
-            None => None,
-            Some(value) => Some(&value.1),
-        }
+    fn get<T: AsRef<str>>(&self, field_name: T) -> Option<&Self::Element> {
+        self.text_fields
+            .get(field_name.as_ref())?
+            .last()
+            .map(|(_s, v)| v)
     }
 
-    fn get_all<'a>(
+    fn get_all<'a, T: AsRef<str>>(
         &'a self,
-        field_name: &'a str,
+        field_name: T,
```

Comment:
Note that the lifetime parameter is not needed for `field_name` it gets elided anyhow with this refactor.